supabase/migrations/10_spec_ext.sql

```diff
@@ -100,12 +112,18 @@ select
   u.email as last_pub_user_email,
   u.full_name as last_pub_user_full_name
 from live_specs l
+inner join internal.user_roles(auth_uid()) on starts_with(l.catalog_name, role_prefix) and capability >= 'read'
 left outer join publication_specs p on l.id = p.live_spec_id and l.last_pub_id = p.pub_id
 left outer join connectors c on c.image_name = l.connector_image_name
 left outer join connector_tags t on c.id = t.connector_id and l.connector_image_tag = t.image_tag
 left outer join internal.user_profiles u on u.user_id = p.user_id;
 ;
-alter view live_specs_ext owner to authenticated;
+-- Using `grant select` is discouraged because it allows the view to query the
```

Comment:
I'm still confused on the motivation here. Is this purely because it has a performance impact (I think)? How much impact?

Or is there an acute correctness issue with how it currently is?

Reply:
The original use of the row-level policy has a strong performance impact, and using the join above instead of the RLS policy makes the query take ~1/3 time it used to take. So queries on this view have got from 2s down to ~600ms.

The approach I took in #742 did not actually have a performance boost, and it was my confusion that I thought it had a performance boost because I had not changed the owner of the view that I was testing on, so I was essentially bypassing the RLS during my tests, hence the boost in performance.

This PR now mixes best of both worlds: we bypass RLS to get the performance boost, but we do an inner join with `user_roles` to make sure we only return items that the user has access to.